mlflow/store/tracking/file_store.py

```diff
@@ -348,17 +354,7 @@ def create_experiment(self, name, artifact_location=None, tags=None):
         self._check_root_dir()
         _validate_experiment_name(name)
         self._validate_experiment_does_not_exist(name)
-        # Get all existing experiments and find the one with largest numerical ID.
-        # len(list_all(..)) would not work when experiments are deleted.
-        experiments_ids = [
-            int(e)
-            for e in (
-                self._get_active_experiments(full_path=False)
-                + self._get_deleted_experiments(full_path=False)
-            )
-            if e.isdigit()
-        ]
-        experiment_id = max(experiments_ids) + 1 if experiments_ids else 0
+        experiment_id = _generate_unique_integer_id()
```

Comment:
Here's the simplest command to reproduce the error:

```
pytest tests/autologging/test_autologging_behaviors_integration.py::test_autolog_respects_silent_mode tests/autologging/test_autologging_client.py::test_client_truncates_param_keys_and_values -s
```

@BenWilson2 @dbczumar 

https://github.com/mlflow/mlflow/blob/d4c88cfdd17b3383b32dc5c674d86059425b16c1/tests/autologging/test_autologging_behaviors_integration.py#L245-L246 seems to be the root cause.


- First, pytest runs `tests/autologging/test_autologging_behaviors_integration.py::test_autolog_respects_silent_mode`.
- Since `tmpdir` exists, `mlflow.set_experiment` skips creating the default experiment.
- `mlflow.set_experiment` creates a new experiment `test_experiment` with an ID generated by `_generate_unique_integer_id`. Let's say the ID is 1234.
- This mutates `_active_experiment_id` in `mlflow/tracking/fluent.py`.
- Next, pytest runs tests in `tests/autologging/test_autologging_client.py::test_client_truncates_param_keys_and_values`. Note `_active_experiment_id` is still 1234.
- In setup, the tracking URI switches to SQLite using `tracking_uri_mock`.
- `mlflow.start_run` is called.
- `mlflow.start_run` searches for an experiment with ID = 1234, but it doesn't exist, then throws an exception.

Reply:
Prior to the change, the generated ID was 0.

Reply:
Setting `mlflow.tracking.fluent._active_experiment_id` to None in teardown of `tests/autologging/test_autologging_behaviors_integration.py::test_autolog_respects_silent_mode` should fix the error.

Reply:
We might want to consider creating the default experiment when the tracking URI for FileStore points to an empty directory.